lib/commands.py

```diff
@@ -322,7 +322,8 @@ def getaddressbalance(self, address):
         server query, results are not checked by SPV.
         """
         sh = bitcoin.address_to_scripthash(address)
-        out = self.network.synchronous_get(('blockchain.scripthash.get_balance', [sh]))
+        minconf = self.config.get('minconf', 1)
+        out = self.network.synchronous_get(('blockchain.scripthash.get_balance', [sh, int(minconf)]))
```

Comment:
This change, as it affects the protocol, should probably be done separately from the rest. In most cases, the client can just filter transactions based on confirmations and calculate whatever, but not if you outsource the request to the server. :)